### PR TITLE
Feat: Use Rich for j2lint outputs and logs

### LINKED ISSUE #13: Add rich for better output
The current plain text formatting could use some better styling.  Several other lint tools leverage [rich](https://github.com/willmcgugan/rich) .

### PR DESCRIPTION
**Pull request by gmuloc**
Fixes #13 

Requires #52 to be merged first

Open for discussions for the current outputs

List rules

Screenshot 2023-01-11 at 22 28 29

No problem

Screenshot 2023-01-11 at 22 29 41

Errors

Screenshot 2023-01-11 at 22 28 55

### CODE REVIEW

File: tests/test_linter/test_error.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def test_to_string(test_issue, verbose, expected):
     Test the string formats for LinterError
     """
 
-    assert test_issue.to_string(verbose) == expected
+    assert str(test_issue.to_string(verbose)) == expected
```
Comment (ClausHolbechArista) [inline tests/test_linter/test_error.py]: we should not accept a non-string response from to_string, so the test here is too forgiving by converting the returned value.

File: tests/test_utils.py [hunk 2]
```diff
@@ -2,20 +2,23 @@
 Tests for j2lint.utils.py
 """
 import os
+
 import pytest
 
+from j2lint.utils import flatten  # load_plugins,
```
Comment (ClausHolbechArista) [inline tests/test_utils.py]: remove the comment and then flatten will end up with the rest

File: tests/test_linter/test_error.py (hunk 1 above)
Comment (gmuloc) [reply tests/test_linter/test_error.py]: so to_string was returning a rich Text and it was simpler to cast it to string. I have changed tthe name of the funciton to `to_rich` which makes more sense and aligned with other changes.

**Comment by ryanmerolle** [comment]
anything blocking this PR?

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Comment by gmuloc** [comment]
Thanks for the nudge @ryanmerolle - can you give us feedback on the current version in devel?

**Comment by ryanmerolle** [comment]
LGTM.  I tested a bunch scenarios.

**Pull request merged**